TodoItemsList component was created (an array of items of a particula…

KGalia opened the pull request:
…r todo is passed in the parameters)

Created TodoItem component (displays item with title and checkbox, one item is passed in parameters) Added id (number), name (string), done (boolean) to the array of items.

KGalia committed:
TodoItemsList component was created (an array of items of a particular todo is passed in the parameters)
Created TodoItem component (displays item with title and checkbox, one item is passed in parameters) Added id (number), name (string), done (boolean) to the array of items.

KGalia committed:
Fix type

ViktoriaStarik commented on vue-project/src/App.vue:
```diff
@@ -3,7 +3,8 @@
 
     <div class="my-style">
-      <todo-list :todos="todos"/>
+      <todo-list :todos="todos"
```
Vue советует использовать PascalCase при использовании своих компонентов https://vuejs.org/guide/essentials/component-basics.html#using-a-component

То есть, нужно писать не `<todo-list/>`, а `<TodoList />`

В документации, возможно, немного непонятно написано. Если коротко, в файлах .js и .vue надо писать TodoList, а в файлах .html todo-list

ViktoriaStarik commented on vue-project/src/App.vue:
```diff
@@ -3,7 +3,8 @@
     <div class="my-style">
-      <todo-list :todos="todos"/>
+      <todo-list :todos="todos"
+                 :key="todos.id"/>
```
Key нужен для оптимизации вью. Имеет смысл использовать его только с v-for и v-if/else, так что удали везде, где не нужно

И у todos нет поля id, насколько я вижу

ViktoriaStarik commented on vue-project/src/App.vue:
```diff
@@ -20,22 +21,47 @@ export default {
+            done: false
+          },
+            {
+              id: new Date().getTime(),
```
Есть вероятность, что итемы создадутся с одинаковыми id, а это плохо для вывода в цикле, потому что вью будет ругаться, что элементы в for с одинаковыми ключами

ViktoriaStarik replied:
Чтобы делать уникальные id, в данном случае можно просто вручную написать их. Но, если хочешь формировать уникальные автоматически, то лучше всего использовать специальную библиотеку, которую в основном все и используют https://www.npmjs.com/package/uuid

KGalia committed:
Configured view router. Created router/index.js file.
The todos variable is moved to the data.js file The logic from App.vue is moved to the MainPage file.
The DetailsPage displays information about Todo: title, list of items

Dontake commented on vue-project/src/components/details/Todo.vue:
выглядит так, что этот файл не нужен и ты можешь описать этот темплейт в файле TodoDetails

Dontake commented on vue-project/src/components/details/TodoDetails.vue:
```diff
@@ -0,0 +1,47 @@
+  components: {Todo},
+  props: {
+    id: String,
+    todo: Object,
```
не понял, зачем этот пропс

Dontake commented on vue-project/src/components/details/TodoDetails.vue:
```diff
@@ -0,0 +1,47 @@
+  data() {
+    return {
+      todosDetails: this.todos,
+      todoItem:{
```
эта переменная не нужна, т.к. мы не будем с ней работать

Dontake commented on vue-project/src/components/details/TodoDetails.vue:
```diff
@@ -0,0 +1,47 @@
+            if (todo.id == id) {
+              this.todoItem = todo;
+            }
+            return todo;
```
можно не возвращать todo, т.к. мы ничего не меняем, просто перебираем массив

Dontake commented on vue-project/src/components/details/TodoDetails.vue:
```diff
@@ -0,0 +1,47 @@
+  methods: {
+    find(id) {
+      this.todosDetails.map((todo) => {
+            if (todo.id == id) {
```
тут нужно строгое сравнение использовать

Dontake commented on vue-project/src/components/details/TodoDetails.vue:
```diff
@@ -0,0 +1,47 @@
+  },
+  methods: {
+    find(id) {
+      this.todosDetails.map((todo) => {
```
вместо map() нужно использовать find()
map используется для изменения элементов массива
я его на созвоне предложил, т.к. не знаю других методов))

Dontake commented on vue-project/src/components/details/TodoDetails.vue:
```diff
@@ -0,0 +1,47 @@
+            return todo;
+          }
+      );
+      return this.todoItem;
```
тут нужно вернуть то что получим через `this.todosDetails.find(id)`

Dontake commented on vue-project/src/components/details/TodoItemDetails.vue:
назови лучше TodoDetailsItem т.к. это элемент именно TodoDetails

Dontake commented on vue-project/src/components/details/TodoItemsListDetails.vue:
и тут тоже назови лучше TodoDetailsItemsList

Dontake commented on vue-project/src/components/Todo.vue:
создай папку main и перенеси в неё Todo и остальные компоненты MainPageView
т.е. путь к ним будет типа такого: components/main/Todo.vue

KGalia committed:
Fix code review
